From 6c7fc3ffcf780a1eafbf83da6c2e54c4b36e588e Mon Sep 17 00:00:00 2001
From: Tobias Fischer <tobias.fischer@qut.edu.au>
Date: Fri, 30 Oct 2026 18:21:36 +1000
Subject: [PATCH 3/7] qE57IO: add PLUGIN_IO_QE57_USE_SYSTEM_LIBE57FORMAT to use
 an installed libE57Format

find_package(E57Format) instead of the bundled extern/libE57Format; off by default.
---
 plugins/core/IO/qE57IO/CMakeLists.txt | 22 ++++++++++++++--------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/plugins/core/IO/qE57IO/CMakeLists.txt b/plugins/core/IO/qE57IO/CMakeLists.txt
index b0e39c0a..9cf2a58d 100644
--- a/plugins/core/IO/qE57IO/CMakeLists.txt
+++ b/plugins/core/IO/qE57IO/CMakeLists.txt
@@ -8,14 +8,20 @@ if ( PLUGIN_IO_QE57 )
 
 	add_subdirectory( include )
 	add_subdirectory( src )
-	add_subdirectory( extern/libE57Format EXCLUDE_FROM_ALL )
-	
-	set_target_properties( E57Format PROPERTIES
-		AUTOMOC OFF
-		AUTORCC OFF
-		AUTOUIC OFF
-	)
-			
+	option( PLUGIN_IO_QE57_USE_SYSTEM_LIBE57FORMAT "Use an installed libE57Format instead of the bundled copy" OFF )
+	if ( PLUGIN_IO_QE57_USE_SYSTEM_LIBE57FORMAT )
+		find_package( E57Format CONFIG REQUIRED )
+	else()
+		add_subdirectory( extern/libE57Format EXCLUDE_FROM_ALL )
+
+		set_target_properties( E57Format PROPERTIES
+			AUTOMOC OFF
+			AUTORCC OFF
+			AUTOUIC OFF
+		)
+	endif()
+
+
 	target_link_libraries( ${PROJECT_NAME} E57Format )
 	
 	if ( USING_STATIC_XERCES )
-- 
2.53.0

